dagstore_test.go

```diff
@@ -80,17 +110,21 @@ func TestRegisterCarV1(t *testing.T) {
 	require.NoError(t, err)
 
 	res := <-ch
-	require.Error(t, res.Error)
-	require.Contains(t, res.Error.Error(), "invalid car version")
```

Comment:
We need a test registering a shard with a corrupted CAR, as we've lost coverage for errors at registration.